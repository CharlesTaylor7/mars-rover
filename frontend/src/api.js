export const rovers = [
  'Curiosity',
  'Opportunity',
  'Spirit',
];

export const fetchCameras = (rover) => fetch(`mars_rover/cameras/?rover=${rover}`).then((res) => res.json());

export const fetchPhotos = ({ rover, camera, method = 'ORM' }) => fetch(
  `mars_rover/photos/${rover}/${camera}/?method=${method}`,
).then((res) => res.json());
