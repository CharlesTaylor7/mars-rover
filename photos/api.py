import requests
import os


def api_key():
    return os.environ['NASA_API_KEY']


def get_photos(rover='', **query):
    query_params = {
        'api_key': api_key(),
        **query,
    }
    payload = requests.get(
        f'https://api.nasa.gov/mars-photos/api/v1/rovers/{rover}/photos',
        params=query_params
    ).json()

    return payload['photos']


def get_rovers():
    query_params = {
        'api_key': api_key(),
    }
    payload = requests.get(
        f'https://api.nasa.gov/mars-photos/api/v1/rovers',
        params=query_params
    ).json()

    return payload['rovers']
