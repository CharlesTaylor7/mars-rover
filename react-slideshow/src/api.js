export const rovers = [
  'Curiosity',
  'Opportunity',
  'Spirit',
];

export const cameras = [{ name: 'FHAZ', full_name: 'Front Hazard Avoidance Camera' }, { name: 'NAVCAM', full_name: 'Navigation Camera' }, { name: 'PANCAM', full_name: 'Panoramic Camera' }, { name: 'MINITES', full_name: 'Miniature Thermal Emission Spectrometer (Mini-TES)' }, { name: 'ENTRY', full_name: 'Entry, Descent, and Landing Camera' }, { name: 'RHAZ', full_name: 'Rear Hazard Avoidance Camera' }];

export const fetchPhotos = ({ rover, camera, method = 'ORM' }) => fetch(
  `photos/${rover}/${camera}/?method=${method}`,
).then((res) => res.json());
