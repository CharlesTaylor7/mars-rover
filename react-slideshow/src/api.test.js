import { fetchPhotos } from './api';

describe('fetchPhotos', () => {
  test('integration test', async () => fetchPhotos({
    rover: 'Curiosity',
    camera: 'FHAZ',
  })
    .then((photos) => expect(photos).toMatchSnapshot()));
});
