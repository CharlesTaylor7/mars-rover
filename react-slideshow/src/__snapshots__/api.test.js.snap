// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPhotos integration test 1`] = `
Array [
  Object {
    "date": "2015-05-30",
    "url": "http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01000/opgs/edr/fcam/FLB_486265257EDR_F0481570FHAZ00323M_.JPG",
  },
  Object {
    "date": "2015-05-30",
    "url": "http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01000/opgs/edr/fcam/FRB_486265257EDR_F0481570FHAZ00323M_.JPG",
  },
]
`;
